core/src/main/scala/org/apache/spark/deploy/rest/SubmitRestProtocolMessage.scala

```diff
@@ -46,9 +46,6 @@ private[rest] abstract class SubmitRestProtocolMessage {
   val action: String = messageType
   var message: String = null
 
-  // For JSON deserialization
-  private def setAction(a: String): Unit = { }
```

Comment:
[SPARK-5388](https://issues.apache.org/jira/browse/SPARK-5388) add this field but it never seems to be called

Reply:
I'm worried that the JSON deserializer needs this, per comment